File: app/src/main/java/com/example/java/android1/java_android_notes/ui/NotesAdapter.java
package com.example.java.android1.java_android_notes.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.java.android1.java_android_notes.DataNote;
import com.example.java.android1.java_android_notes.DataNoteSource;
import com.example.java.android1.java_android_notes.R;

public class NotesAdapter extends RecyclerView.Adapter<ViewHolder> {

    private final LayoutInflater inflater;
    private final DataNoteSource mDataNoteSource;
    private OnItemClickListener mOnItemClickListener;
    private OnItemCreateContextMenuListener mOnItemCreateContextMenuListener;

    public NotesAdapter(LayoutInflater inflater, DataNoteSource dataNoteSource) {
        this.inflater = inflater;
        this.mDataNoteSource = dataNoteSource;
    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.mOnItemClickListener = onItemClickListener;
    }

    public void setOnItemCreateContextMenuListener(OnItemCreateContextMenuListener onItemCreateContextMenuListener) {
        this.mOnItemCreateContextMenuListener = onItemCreateContextMenuListener;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.note_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        DataNote dataNote = mDataNoteSource.getItem(position);
        holder.itemView.setOnClickListener((view) -> {
            if (mOnItemClickListener != null) {
                mOnItemClickListener.onItemClickListener(view, position);
            }
        });
        holder.itemView.setOnCreateContextMenuListener((contextMenu, view, contextMenuInfo) ->
                mOnItemCreateContextMenuListener.onItemCreateContextMenuListener(contextMenu, view, contextMenuInfo, position));
        holder.fillCard(dataNote);
    }

    @Override
    public int getItemCount() {
        return mDataNoteSource.getDataNoteCount();
    }
}

File: app/src/main/java/com/example/java/android1/java_android_notes/view/main/NotesAdapter.kt
package com.example.java.android1.java_android_notes.view.main

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.CompoundButton.OnCheckedChangeListener
import androidx.recyclerview.widget.RecyclerView
import com.example.java.android1.java_android_notes.databinding.NoteItemBinding
import com.example.java.android1.java_android_notes.model.DataNote

class NotesAdapter(
    private val onLongClickListener: ((DataNote) -> Unit)?,
    private val onItemClickListener: ((DataNote) -> Unit)?,
    private val onCheckedChangeListener: (DataNote, Boolean) -> Unit
) : RecyclerView.Adapter<ViewHolder>() {

    private val notesData: MutableList<DataNote> = mutableListOf()

    @SuppressLint("NotifyDataSetChanged")
    fun setNotesList(notes: List<DataNote>) {
        notesData.clear()
        notesData.addAll(notes)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
        ViewHolder(
            NoteItemBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        ).apply {
            itemView.setOnClickListener {
                onItemClickListener?.invoke(notesData[layoutPosition])
            }
            itemView.setOnLongClickListener {
                onLongClickListener?.invoke(notesData[layoutPosition])
                return@setOnLongClickListener true
            }
        }

    override fun getItemCount(): Int = notesData.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.cardBind(dataNote = notesData[position], onCheckedChangeListener)
    }

}

class ViewHolder(private val binding: NoteItemBinding) : RecyclerView.ViewHolder(binding.root) {

    fun cardBind(dataNote: DataNote, onCheckedChangeListener: (DataNote, Boolean) -> Unit) {
        binding.cardItemTitle.text = dataNote.noteTitle
        binding.cardItemDescription.text = dataNote.noteDescription
        binding.cardItemDate.text = dataNote.noteDate
        binding.cardItemFavorite.apply {
            isChecked = dataNote.noteFavorite ?: false
            setOnCheckedChangeListener { _, isChecked ->
                onCheckedChangeListener.invoke(dataNote, isChecked)
            }
        }
    }

}